### ISSUE TITLE
[BUG]Deleted Nodes reappear after some time, if deleted from https://<cluster-hostname>/dashboard/c/<cluster_id>/explorer/node

### ISSUE THREAD

**Issue by rronneburger** [rancher/rancher#43680]
**Rancher Server Setup**
- Rancher version: 2.7.9
- Installation option (Docker install/Helm Chart):
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): RKE2 v1.26.10
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: v1.26.10
- Cluster Type (Local/Downstream): Custom (VMware virtual machines)
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): Custom

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom): Admin

**Describe the bug**
I deleted 2 Nodes (after draining them) from https://<cluster-hostname>/dashboard/c/<cluster_id>/explorer/node . Then I waited until they where gone and both the explorer/node and the cluster-management page showed the cluster as "Active" and displayed the correct number of nodes. I also checked with kubectl, everything seemed to be fine. 

I then created two new VMs from scratch and added them to the cluster. Again both pages + kubectl showed the correct number of nodes and showed everything as "Active". After some hours cluster management displayed:

Configuring worker node(s) custom-c521de9113a2,custom-c681ffc65509 

The displayed node-ids are the IDs of the deleted nodes. Now the deleted nodes show up again in https://<cluster-hostname>/dashboard/c/_/manager/provisioning.cattle.io.cluster/fleet-default/<cluster-name>#machine-pools , together with the newly provisioned nodes that have replaced them.

If I delete one of the (already deleted) nodes, both the old node and the new node (with the same name) are gone.

If I delete the node via cluster-manager initially, everything seems to work fine. To check if it is really gone one can use:

kubectl get Machines -A

on the rancher-cluster. If the count there differs from the count in kubectl, then there is a problem.

**To Reproduce**
Delete a custom Node from https://<cluster-hostname>/dashboard/c/<cluster_id>/explorer/node , wait until it's gone, then reprovision it as a new VM and register it with the Registration Command displayed by Rancher.

**Result**
After some time 2 nodes with the same name are displayed on dashboard/c/_/manager/provisioning.cattle.io.cluster/fleet-default/<cluster-name>#machine-pools and the status in cluster-management switches to "Configuring worker node(s) custom-XYZ" and remains there.

**Expected Result**
The node should be gone in all places in Rancher-UI and in "kubectl get Machines -A".

**Additional context**
The VMs (for both the rancher-cluster and the downstream cluster) use ubuntu 22.04 patched to the latest version.

**Comment by Oats87** [comment]
@rronneburger 

a couple of clarifying questions if you don't mind me asking

1. Were the names of the corresponding machine objects for the two nodes you deleted custom-c521de9113a2, and custom-c681ffc65509 ?
2. Are you shutting down/deleting the VMs that you removed from the cluster?
3. To confirm, you are deleting the K8s node object right? I would recommend deleting the machine object and letting the CAPI controllers delete your node object.

**Comment by rronneburger** [comment]
> @rronneburger
> 
> a couple of clarifying questions if you don't mind me asking
> 
>     1. Were the names of the corresponding machine objects for the two nodes you deleted custom-c521de9113a2, and custom-c681ffc65509 ?

Yes, these where the internal names. The hostnames where different (kube-<name>04 and -05).

>     2. Are you shutting down/deleting the VMs that you removed from the cluster?

Yes, the VMs where deleted and then redeployed.

>     3. To confirm, you are deleting the K8s node object right? I would recommend deleting the machine object and letting the CAPI controllers delete your node object.

I deleted the node from the rancher-gui as described ( https:///dashboard/c/<cluster_id>/explorer/node ), the same place where i could drain it before and where rancher offers me the delete action.

**Comment by Oats87** [comment]
I think what may be happening is your order of operations is causing the CAPI machine objects to stick around. Coincidentally, you can see a (potentially related) issue here: https://github.com/rancher/rancher/issues/43686 that details the problem.

More or less, if you delete the Kubernetes node object that correlates to a cluster-api machine object, the cluster-api core (machine) controller is unable to clean up the object as the node reference is invalid. I am still investigating a safe workaround for this.
